src/main/java/com/sprint/mission/discodeit/repository/file/FileChannelRepository.java

```diff
@@ -0,0 +1,90 @@
+package com.sprint.mission.discodeit.repository.file;
+
+import com.sprint.mission.discodeit.dto.entity.Channel;
+import com.sprint.mission.discodeit.dto.form.ChannelUpdateDto;
+import com.sprint.mission.discodeit.repository.ChannelRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+
+
+@Slf4j
+public class FileChannelRepository implements ChannelRepository {
+    private static final String FILE_PATH = "temp/channels-obj.dat";
+    private final Map<UUID, Channel> data=new HashMap<>();
+
+    @Override
+    public Channel createChannel(UUID id, Channel channel) {
+        data.put(id, channel);
+        save();
+        return channel;
+    }
+
+    @Override
+    public void updateChannel(UUID id, ChannelUpdateDto channelUpdateDto) {
+        Channel findChannel = data.get(id);
+        if(findChannel.getChannelGroup().equals("PUBLIC")) {
```

Comment:
이 코드는 항상 false가 뜨는데요,  

enum은 보통 equals 대신 == 비교를 사용합니다.

```suggestion
        if(findChannel.getChannelGroup() == ChannelGroup.PUBLIC) {
```

물론 책임을 던지면 더 좋구요.  
```suggestion
        if(findChannel.isPublic()) {
```